numba/core/base.py

```diff
@@ -556,7 +556,16 @@ def get_function(self, fn, sig, _firstcall=True):
         except NotImplementedError:
             pass
         if isinstance(fn, types.Type):
+            # It's a type instance => try to find a call overload
+            try:
+                return self.get_function((type(fn), '__call__'), sig)
+            except NotImplementedError:
+                # Raise exception for the type instance, for a better error message
+                pass
+
             # It's a type instance => try to find a definition for the type class
+            # TODO: this is weird, why are both static and non-static calls in
+            # the same place.
             try:
                 return self.get_function(type(fn), sig)
             except NotImplementedError:
```

Comment:
I don't really understand where this code path is hit. If I remove it, then I get errors like `No definition for lowering class(uint8)(Literal[int](1),) -> uint8`